lasy/profiles/speckle_profile.py

```diff
@@ -0,0 +1,519 @@
+import numpy as np
+from scipy.constants import c
+
+from .profile import Profile
+
+
+def gen_gaussian_time_series(t_num, dt, fwhm, rms_mean):
+    """Generate a discrete time series that has gaussian power spectrum.
+
+    Parameters
+    ----------
+    t_num: number of grid points in time
+    fwhm: full width half maximum of the power spectrum
+    rms_mean: root-mean-square average of the spectrum
+
+    Returns
+    -------
+    temporal_amplitude: a time series array of complex numbers with shape [t_num]
+    """
+    if fwhm == 0.0:
+        temporal_amplitude = np.zeros((2, t_num))
+    else:
+        omega = np.fft.fftshift(np.fft.fftfreq(t_num, d=dt))
+        psd = np.exp(-np.log(2) * 0.5 * np.square(omega / fwhm * 2 * np.pi))
+        spectral_amplitude = np.array(psd) * (
+            np.random.normal(size=t_num) + 1j * np.random.normal(size=t_num)
+        )
+        temporal_amplitude = np.fft.ifftshift(
+            np.fft.fft(np.fft.fftshift(spectral_amplitude))
+        )
+        temporal_amplitude *= rms_mean / np.sqrt(
+            np.mean(np.square(np.abs(temporal_amplitude)))
+        )
+    return temporal_amplitude
+
+
+class SpeckleProfile(Profile):
+    r"""
+    Derived class for the profile of a speckled laser pulse.
+
+    Speckled lasers are used to mitigate laser-plasma interactions in fusion and ion acceleration contexts.
+    More on the subject can be found in chapter 9 of `Introduction to Laser-Plasma Interactions <https://link.springer.com/book/10.1007/978-3-031-23424-8>`__.
+    A speckled laser beam is a laser that is deliberately divided transversely into several beamlets in the near-field.
+    This is done with a near-field phase plate divided into
+    The phase plate provides a different phase to each beamlet, which then propagate incoherently and combine in the far field.
+
+    The electric field corresponds to:
+
+    .. math::
+
+        \begin{aligned}
+        E_u(\boldsymbol{x}_\perp,t) &= Re\left[ E_0
+        \sum_{j=1}^{N_{bx}\times N_{by}} A_j
+        {\rm sinc}\left(\frac{\pi D_xx}{\lambda_0 f}\right)
+        {\rm sinc}\left(\frac{\pi D_yy}{\lambda_0 f}\right)
+        \right.
+        \\
+        & \left. \times\exp\left(i\boldsymbol{k}_{\perp,j}\cdot\boldsymbol{x}_\perp
+        + i\phi_{{\rm RPP/CPP},j}+i\psi_{{\rm SSD},j}(t)\right) \times p_u
+        \right]
+        \end{aligned}
+
+    where :math:`u` is either :math:`x` or :math:`y`, :math:`p_u` is
+    the polarization vector, :math:`Re` represent the real part, and
+    :math:`\boldsymbol{x}_\perp=(x,y)` is the transverse coordinate (orthogonal
+    to the propagation direction).
+    Several quantities are computed internally to the code depending on the
+    method of smoothing chosen, including the beamlet amplitude :math:`A_j`,
+    the beamlet wavenumber at focus :math:`k_{\perp,j}`,
```

Comment:
The wavenumber is not necessarily different *at focus* than at other positions, no?
```suggestion
    the beamlet wavenumber :math:`k_{\perp,j}`,
```